### PR TITLE
feat: add scope completion support

### LINKED ISSUE #3: support scope of conventional commits
Conventional commit lets you add a scope (e.g. `test(scope):`) but the plugin suggests the conventional commits keywords again inside the braces.

See [4. of conventional commits spec](https://www.conventionalcommits.org/en/v1.0.0/#specification)

### PR DESCRIPTION
**Pull request by benner**
Parses the last 200 commits to extract previously used scopes, offering them as completions when the cursor is inside type(...).

Close https://github.com/disrupted/blink-cmp-conventional-commits/issues/3

### CODE REVIEW

File: lua/blink-cmp-conventional-commits/init.lua [hunk 1]
```diff
@@ -10,11 +10,42 @@
...
+        label = scope,
+        insertText = scope,
+        kind = require('blink.cmp.types').CompletionItemKind.Value,
+    }
+end
+
+---@param count integer
+---@return string[]
+local function discover_scopes(count)
+    local output = vim.fn.system(
```
Comment (disrupted) [inline lua/blink-cmp-conventional-commits/init.lua]: can we use `vim.system`? I would like to avoid `vim.fn` and blocking IO in general

File: lua/blink-cmp-conventional-commits/init.lua [hunk 2]
```diff
@@ -10,11 +10,42 @@
...
+        'git log --no-merges --oneline -' .. count
+    )
+    if vim.v.shell_error ~= 0 then
+        return {}
+    end
+    local seen = {}
+    for scope in output:gmatch '%w+%((.-)%)' do
```
Comment (disrupted) [inline lua/blink-cmp-conventional-commits/init.lua]: I believe we'll need some tests to avoid edge cases

File: README.md [hunk 3]
```diff
@@ -43,6 +43,30 @@ example using [Lazy](https://github.com/folke/lazy.nvim) plugin manager
...
 
+## Scope Completion
+
+The plugin automatically discovers scopes used in the repository by
+scanning recent non-merge commits. When you type `feat(` or any other
+type followed by an opening parenthesis, completions for previously
+used scopes will appear.
+
+Scopes are discovered at plugin load time from `git log`. The number
+of commits to scan can be configured via `git_log_count` (default: 200):
```
Comment (disrupted) [inline README.md:54]: any specific reason why you went with 200 for the default?

**Review by disrupted — COMMENTED** [review]
Thanks a lot for this! I am not near a computer right now but will give it a try later. How noticeable is the delay when it loads the git commit log?

File: README.md (hunk 3 above)
Comment (benner) [reply README.md:54]: No specific reason. I just wanted to have some limit for speed and safety.

File: lua/blink-cmp-conventional-commits/init.lua (hunk 1 above)
Comment (benner) [reply lua/blink-cmp-conventional-commits/init.lua]: changed

File: lua/blink-cmp-conventional-commits/init.lua (hunk 2 above)
Comment (benner) [reply lua/blink-cmp-conventional-commits/init.lua]: added guards for empty (`feat():`) or with spaces (`feat( foo ): bar`) cases.

File: lua/blink-cmp-conventional-commits/init.lua [hunk 4]
```diff
@@ -10,11 +10,45 @@
...
+        kind = require('blink.cmp.types').CompletionItemKind.Value,
+    }
+end
+
+---@param count integer
+---@return string[]
+local function discover_scopes(count)
+    local result =
+        vim.system({ 'git', 'log', '--no-merges', '--oneline', '-' .. count })
+            :wait()
```
Comment (disrupted) [inline lua/blink-cmp-conventional-commits/init.lua]: this is still blocking the main thread. We would need to use the callback-based async variant

File: lua/blink-cmp-conventional-commits/init.lua (hunk 2 above)
Comment (benner) [reply lua/blink-cmp-conventional-commits/init.lua]: also found more cases and covered them too.

File: lua/blink-cmp-conventional-commits/init.lua (hunk 4 above)
Comment (benner) [reply lua/blink-cmp-conventional-commits/init.lua]: done

File: lua/blink-cmp-conventional-commits/init.lua [hunk 5]
```diff
@@ -10,11 +10,52 @@
...
+                local scope = line:match '^%x+ %w+%((.-)%)'
+                if scope then
+                    scope = scope:match '^%s*(.-)%s*$'
+                    if scope ~= '' then
+                        seen[scope] = true
+                    end
+                end
+            end
+            local scopes = vim.tbl_keys(seen)
+            table.sort(scopes)
```
Comment (disrupted) [inline lua/blink-cmp-conventional-commits/init.lua]: could you explain why it is necessary to sort the table? wouldn't it be more natural for suggestions to be from newest (most recent) to oldest?

**Review by disrupted — COMMENTED** [review]
left one clarifying question, other than that LGTM

File: lua/blink-cmp-conventional-commits/init.lua (hunk 5 above)
Comment (benner) [reply lua/blink-cmp-conventional-commits/init.lua]: I'll change

**Review by disrupted — APPROVED** [review]
thanks for the contribution!
